src/cursor/cur_file.c

```diff
@@ -8,6 +8,13 @@
 
 #include "wt_internal.h"
 
+/*
+ * Define functions that increment histogram statistics for cursor read and
+ * write operations latency.
+ */
+WT_STAT_USECS_HIST_INCR_FUNC(opread, perf_hist_opread_latency, 100)
+WT_STAT_USECS_HIST_INCR_FUNC(opwrite, perf_hist_opwrite_latency, 100)
```

Comment:
@sulabhM this looks OK to me, but it's a new pattern in WiredTiger source base, I think so I'm going to ask @sueloverso to review as well.

Reply:
Hmm, I was hoping we'd get 4 functions down to one macro. I didn't realize the very slightly different ranges. Would we want to reconsider those ranges to be able to collapse them?  While I know the units are different, usecs versus msecs, I do think that looking at the stats in t2 it will be easier to only keep a consistent set of values in our head, or add an arg to the macro that lets them adjust some?

Reply:
I talked to Sulabh about this today, and I'd talked to Dave and the buckets when he was making the change originally. I think having two sets of buckets is the best solution. One for slower ops and one for faster ops. Attempting to share the same set across all histogram stats leads to having a lot more buckets per histogram which has a real space cost in our data structures.
We couldn't figure out a clean way to keep two sets of buckets with a single macro (we pretty much ended up passing in a set of buckets to each macro call).
Medium term I'd like to have this code auto generated by dist/stat.py,  but I think we are more likely to get that right if we do it for the next set of histogram stats we add..
If you had a concrete change in mind here it would be great if you can push a change.

Reply:
Thanks for the info on the histogram ranges. Go ahead and merge.